corehq/apps/geospatial/reports.py

```diff
@@ -136,6 +144,12 @@ def _build_query(self):
         query = apply_geohash_agg(query, case_property, precision)
         return query
 
+    def _get_geo_point(self, case):
+        case_obj = wrap_case_search_hit(case)
+        geo_case_property = get_geo_case_property(case_obj.domain)
```

Comment:
nit: I know `get_geo_case_property` is cached. Though since it will be called for all cases which could be 10k I assume, may be we could just add a cached_property on the class itself. 
In fact, Since all cases are expected to be of the same domain, we should fetch it only once?

Reply:
This was just copied from the existing functionality, however I agree that this only needs to be called once. I'll update this.

Reply:
[9166c5b](https://github.com/dimagi/commcare-hq/pull/33610/commits/9166c5b730b6b68070dd3c3ff9b2ac5d62e6d235)

Reply:
> This was just copied from the existing functionality,

Then this probably needs to be fixed at multiple places. :) 
(not in this PR)